src/Workspaces/Core/Portable/CodeGeneration/CodeGenerationHelpers.cs

```diff
@@ -102,6 +103,60 @@ public static bool IsSpecialType(ITypeSymbol type, SpecialType specialType)
             return type != null && type.SpecialType == specialType;
         }
 
+        /// <summary>
+        /// Indicates whether the specified attribute is
+        /// <see cref="AllowNullAttribute"/>,
+        /// <see cref="MaybeNullAttribute"/> or
+        /// <see cref="MaybeNullWhenAttribute"/>.
+        /// </summary>
+        public static bool IsAllowNullOrMaybeNullAttribute(AttributeData attribute) =>
+            IsFlowAnalysisNamespace(attribute.AttributeClass.ContainingNamespace) &&
+                attribute.AttributeClass.Name switch
+                {
+                    nameof(AllowNullAttribute) => true,
+                    nameof(MaybeNullAttribute) => true,
+                    nameof(MaybeNullWhenAttribute) => true,
+                    _ => false
+                };
+
+        /// <summary>
+        /// Indicates whether the specified attribute is
+        /// <see cref="DisallowNullAttribute"/>,
+        /// <see cref="NotNullAttribute"/>,
+        /// <see cref="NotNullWhenAttribute"/> or
+        /// <see cref="NotNullIfNotNullAttribute"/>.
+        /// </summary>
+        public static bool IsDisallowNullOrNotNullAttribute(AttributeData attribute) =>
+            IsFlowAnalysisNamespace(attribute.AttributeClass.ContainingNamespace) &&
+                attribute.AttributeClass.Name switch
+                {
+                    nameof(DisallowNullAttribute) => true,
+                    nameof(NotNullAttribute) => true,
+                    nameof(NotNullWhenAttribute) => true,
+                    nameof(NotNullIfNotNullAttribute) => true,
+                    _ => false
+                };
+
+        public static bool IsNullableFlowAnalysisAttribute(AttributeData attribute) =>
+            IsFlowAnalysisNamespace(attribute.AttributeClass.ContainingNamespace) &&
+                attribute.AttributeClass.Name switch
+                {
+                    nameof(AllowNullAttribute) => true,
+                    nameof(DisallowNullAttribute) => true,
+                    nameof(MaybeNullAttribute) => true,
+                    nameof(NotNullAttribute) => true,
+                    nameof(MaybeNullWhenAttribute) => true,
+                    nameof(NotNullWhenAttribute) => true,
+                    nameof(NotNullIfNotNullAttribute) => true,
+                    _ => false
```

Comment:
this looks duplicative.  can this not be: `IsAllowNullOrMaybeNullAttribute || IsDisallowNullOrNotNullAttribute`?